File: lua/HelloWorld_main.lua
#!/usr/bin/env lua

local function safe_print(msg)
            local success, err = pcall(function()
                        print(msg)
                        io.flush()
                        if not io.output():write() then
                                    error("Output stream error")
                        end
            end)
            if not success then
                        error("Failed to print: " .. tostring(err))
            end
end

local ok, err = pcall(safe_print, "Hello World!")
if not ok then
            io.stderr:write("Output error: ", err, "\n")
            os.exit(1)
end

File: go/HelloWorld_main.go
package main
import (
            "fmt"
            "os"
)

func safePrintHello() error {
            _, err := fmt.Println("Hello World!")
            if err != nil {
                        return fmt.Errorf("output error: %w", err)
            }
            return nil
}

func main() {
            if err := safePrintHello(); err != nil {
                        fmt.Fprintln(os.Stderr, err)
                        os.Exit(1)
            }
}
